pay/lib/src/pay.dart

```diff
@@ -72,12 +49,21 @@ class Pay {
   /// This method wraps the [showPaymentSelector] method in the platform
   /// interface, and opens the payment selector for the [provider] of choice,
   /// with the [paymentItems] in the price summary.
-  Future<Map<String, dynamic>> showPaymentSelector({
-    PayProvider? provider,
-    required List<PaymentItem> paymentItems,
-  }) async {
-    await _assetInitializationFuture;
-    final configuration = _findConfig(provider);
-    return _payPlatform.showPaymentSelector(configuration, paymentItems);
+  Future<Map<String, dynamic>> showPaymentSelector(
+    PayProvider provider,
+    List<PaymentItem> paymentItems,
+  ) async {
+    throwIfProviderIsNotDefined(provider);
+    return _payPlatform.showPaymentSelector(
+        _configurations[provider]!, paymentItems);
+  }
+
+  /// Verifies that the selected provider has been previously configured or
+  /// throws otherwise.
+  void throwIfProviderIsNotDefined(PayProvider provider) {
+    if (!_configurations.containsKey(provider)) {
+      throw Exception(
```

Comment:
**[non-blocking]** Would be nice here to have a custom throwable for this case should the consumer want to handle accordingly. (e.g. `InvalidPaymentProvider`)

Reply:
Good idea, will add

Reply:
Added in commit f52026e.
PTAL @niccorder

Reply:
Looks great – thanks Jose!